database/gormdb.go

```diff
@@ -375,7 +376,31 @@ func (db *GormDB) CreateSubmission(submission *pb.Submission) error {
 	if assignment+group != 2 {
 		return gorm.ErrRecordNotFound
 	}
-	return db.conn.Create(submission).Error
+
+	// If a submission for the given assignment and student/group already exists, update it.
+	// Otherwise create a new submission record
+	query := &pb.Submission{
+		AssignmentID: submission.GetAssignmentID(),
```

Comment:
Do we need to create a new Submission object here? Can’t we just reuse the submission object?

Reply:
The submission object includes fields with test results that the query will also attempt to match. This way, when updating a submission, the old submission would be ignored, as these fields would be different. 
And we just want to check whether a submission for the given lab and user/group exists, no matter what's in the other fields.

Reply:
Fight. This is reasonable. But maybe add a comment to explain this, since I’m bound to ask again :)

Reply:
Just did.